File: android/app/src/main/java/com/ahmedkhaled/onroute/model/POIResult.kt
package com.ahmedkhaled.onroute.model

import androidx.compose.ui.graphics.Color
import com.google.android.gms.maps.model.LatLng
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class POIResult(
    val placeId: String,
    val name: String,
    val address: String,
    val lat: Double,
    val lng: Double,
    val detourSeconds: Int,
    val detourFormatted: String,
    val rating: Double,
    val userRatingCount: Int,
    val isOpenNow: Boolean,
    val priceLevel: String? = null,
    val phoneNumber: String? = null,
    val todayHours: String? = null,
    val types: List<String>,
    val photoReference: String? = null
) {
    val latLng: LatLng get() = LatLng(lat, lng)

    val priceLevelDisplay: String? get() = when (priceLevel) {
        "PRICE_LEVEL_INEXPENSIVE" -> "$"
        "PRICE_LEVEL_MODERATE" -> "$$"
        "PRICE_LEVEL_EXPENSIVE" -> "$$$"
        "PRICE_LEVEL_VERY_EXPENSIVE" -> "$$$$"
        else -> null
    }

    val detourColor: Color
        get() {
            val minutes = detourSeconds / 60
            return when {
                minutes < 3 -> Color(0xFF33B249)
                minutes < 7 -> Color(0xFFE6B312)
                minutes < 15 -> Color(0xFFF28018)
                else -> Color(0xFFE63333)
            }
        }
}

@JsonClass(generateAdapter = true)
data class RouteInfo(
    val encodedPolyline: String,
    val durationSeconds: Int,
    val distanceMeters: Int
)

@JsonClass(generateAdapter = true)
data class StopResults(
    val query: String,
    val results: List<POIResult>
)

@JsonClass(generateAdapter = true)
data class SearchResponse(
    val results: List<POIResult>,
    val stops: List<StopResults>? = null,
    val route: RouteInfo,
    val cached: Boolean? = null
)

@JsonClass(generateAdapter = true)
data class SearchRequest(
    val origin: LatLngBody,
    val destination: LatLngBody,
    val query: String,
    val queries: List<String>? = null,
    val maxDetourMinutes: Int? = null,
    val openNow: Boolean = true,
    val travelMode: String = "DRIVE"
)

@JsonClass(generateAdapter = true)
data class LatLngBody(val lat: Double, val lng: Double)

File: ios/Detour/Models/POIResult.swift
import Foundation
import CoreLocation

struct POIResult: Identifiable, Codable, Hashable {
    let placeId: String
    let name: String
    let address: String
    let lat: Double
    let lng: Double
    let detourSeconds: Int
    let detourFormatted: String
    let rating: Double
    let userRatingCount: Int
    let isOpenNow: Bool
    let priceLevel: String?
    let phoneNumber: String?
    let todayHours: String?
    let types: [String]
    let photoReference: String?

    var id: String { placeId }

    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: lat, longitude: lng)
    }

    var priceLevelDisplay: String? {
        switch priceLevel {
        case "PRICE_LEVEL_INEXPENSIVE": return "$"
        case "PRICE_LEVEL_MODERATE": return "$$"
        case "PRICE_LEVEL_EXPENSIVE": return "$$$"
        case "PRICE_LEVEL_VERY_EXPENSIVE": return "$$$$"
        default: return nil
        }
    }

    var detourColor: DetourColor {
        let minutes = detourSeconds / 60
        if minutes < 3 { return .green }
        if minutes < 7 { return .yellow }
        if minutes < 15 { return .orange }
        return .red
    }

    enum DetourColor: String {
        case green, yellow, orange, red
    }
}

struct StopResults: Codable {
    let query: String
    let results: [POIResult]
}

struct SearchResponse: Codable {
    let results: [POIResult]
    let stops: [StopResults]?
    let route: RouteInfo
}

struct RouteInfo: Codable {
    let encodedPolyline: String
    let durationSeconds: Int
    let distanceMeters: Int
}
